sdk/platform/http_client/curl/inc/curl_client.hpp

```diff
@@ -10,30 +10,74 @@ class CurlClient
 {
 private:
   azure::core::http::Request& m_request;
-  CURL* m_p_curl;
+  // for every client instance, create a default response
+  azure::core::http::Response m_response;
```

Comment:
This calls the default ctor, right?
If so, this is not really the best design because we now have an invalid Response living in the system.
It would be better to make this a pointer and then assign to it once we have created a valid Response.
Responses have to live in the heap anyway because they have to outlive the pipeline.